tests/e2e/module_consistency_test.go

```diff
@@ -0,0 +1,95 @@
+package e2e_test
+
+import (
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/kyma-project/lifecycle-manager/api/shared"
+	"github.com/kyma-project/lifecycle-manager/api/v1beta2"
+
+	. "github.com/kyma-project/lifecycle-manager/pkg/testutils"
+)
+
+var _ = Describe("Module Keep Consistent After Deploy", Ordered, func() {
+	kyma := NewKymaWithSyncLabel("kyma-sample", "kcp-system", v1beta2.DefaultChannel,
+		v1beta2.SyncStrategyLocalSecret)
+	module := NewTemplateOperator(v1beta2.DefaultChannel)
+	moduleCR := NewTestModuleCR(remoteNamespace)
+
+	InitEmptyKymaBeforeAll(kyma)
+	CleanupKymaAfterAll(kyma)
+
+	Context("Given SKR Cluster", func() {
+		It("When Kyma Module is enabled on SKR Kyma CR", func() {
+			Eventually(EnableModule).
+				WithContext(ctx).
+				WithArguments(runtimeClient, defaultRemoteKymaName, remoteNamespace, module).
+				Should(Succeed())
+		})
+
+		It("Then Module Resources are deployed on SKR cluster", func() {
+			By("And Module CR exists")
+			Eventually(ModuleCRExists).
+				WithContext(ctx).
+				WithArguments(runtimeClient, moduleCR).
+				Should(Succeed())
+			By("And Module Operator Deployment is ready")
+			Eventually(DeploymentIsReady).
+				WithContext(ctx).
+				WithArguments(runtimeClient, "template-operator-controller-manager", TestModuleResourceNamespace).
+				Should(Succeed())
+
+			By("And KCP Kyma CR is in \"Ready\" State")
+			Eventually(KymaIsInState).
+				WithContext(ctx).
+				WithArguments(kyma.GetName(), kyma.GetNamespace(), controlPlaneClient, shared.StateReady).
+				Should(Succeed())
+		})
+		It("Then synced module resources keep consistent by verify resourceVersion not update", func() {
+			manifest, err := GetManifest(ctx, controlPlaneClient, kyma.GetName(), kyma.GetNamespace(),
+				module.Name)
+			Expect(err).Should(Succeed())
+			for _, resource := range manifest.Status.Synced {
+				objectKey := client.ObjectKey{Name: resource.Name, Namespace: resource.Namespace}
+				gvk := schema.GroupVersionKind{Group: resource.Group, Version: resource.Version, Kind: resource.Kind}
+				resourceInCluster, err := GetCR(ctx, runtimeClient,
+					objectKey,
+					gvk)
+				Expect(err).Should(Succeed())
+				Consistently(IsResourceVersionSame).
+					WithContext(ctx).
+					WithArguments(runtimeClient, objectKey, gvk,
+						resourceInCluster.GetResourceVersion()).Should(BeTrue())
+			}
+		})
+
+		It("Then synced module resources keep consistent by revert modification", func() {
```

Comment:
I think this statement is not clear enough, and also this is should start with an And and be moved to the previous It block

Reply:
Do you have a better suggestion for this? 😄 

Regarding this BDD syntax, I think we need to clarify here, it's not strictly defined that one test case only allow one `Then` keyword, I agree with `And` it makes the statement more fluidly, but multiple `Then` in most case is more strateforward. Check the `And, But` section in Gherkin [document](https://cucumber.io/docs/gherkin/reference/)

And same as the reason for this block, it checks specifically resource get reset after user modified, it's not good mix with first `It` block, they address different concern.

Reply:
It's not strict but IMO if you read the whole test as what's generated in the report then you'd expect all the Thens are concatenated with And, except if there's a new When (or an action that took place in order for this result to happen).